wave_benchmark/include/wave/benchmark/trajectory_compare.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef WAVE_BENCHMARK_TRAJECTORY_COMPARE_HPP
+#define WAVE_BENCHMARK_TRAJECTORY_COMPARE_HPP
+
+#include <iostream>
+#include <fstream>
+
+#include "pose_measurement.hpp"
+#include "wave/containers/measurement_container.hpp"
+#include "wave/geometry/rotation.hpp"
+
+namespace wave {
+
+class TrajectoryCompare {
+ public:
+    void reset();
+    void push_truth(Pose pose, const TimeType time);
+    void push_measurement(Pose pose, const TimeType time);
+    void calculate_error();
+    void output_csv(std::string path, std::string filename);
+    MeasurementContainer<PoseMeasurement> ground_truth;
+    MeasurementContainer<PoseMeasurement> measurements;
+    MeasurementContainer<PoseMeasurement> error;
+
+ private:
+    const int ground_truth_key = 0;
```

Comment:
If you have a small set of possible values, not representing a quantity, use an enum.